### PR TITLE
support fetch user info based on query type

### PR DESCRIPTION
**Pull request by lsy1990**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #
 
**Special notes for your reviewer**:

### CODE REVIEW

File: src/pkg/oauth2x/oauth2x.go [hunk 1]
```diff
@@ -167,15 +170,33 @@ func exchangeUser(code string) (*CallbackOutput, error) {
 	}, nil
 }
 
-func getUserInfo(userInfoAddr, accessToken string) ([]byte, error) {
-	r, err := http.NewRequest("GET", userInfoAddr, nil)
-	if err != nil {
-		return nil, err
+func getUserInfo(userInfoAddr, accessToken string, TranTokenMethod int) ([]byte, error) {
+	logger.Errorf("getUserInfo: %s", userInfoAddr)
```
Comment (UlricQin) [inline src/pkg/oauth2x/oauth2x.go:174]: 这里为啥要打印error呢

File: etc/webapi.conf [hunk 2]
```diff
@@ -193,6 +193,7 @@ RedirectURL = "http://127.0.0.1:18000/callback/oauth"
 SsoAddr = "https://sso.example.com/oauth2/authorize"
 TokenAddr = "https://sso.example.com/oauth2/token"
 UserInfoAddr = "https://api.example.com/api/v1/user/info"
+TranTokenMethod = 0
```
Comment (UlricQin) [inline etc/webapi.conf:196]: 感觉数字表意不明确，看看能否用字符串来表示。比如 header、querystring、formdata

File: src/pkg/oauth2x/oauth2x.go (hunk 1 above)
Comment (lsy1990) [reply src/pkg/oauth2x/oauth2x.go:174]: 日志忘记删除了

File: etc/webapi.conf (hunk 2 above)
Comment (lsy1990) [reply etc/webapi.conf:196]: 可以

**Pull request merged**